libs/remix-core-plugin/src/lib/compiler-content-imports.ts

```diff
@@ -115,66 +116,58 @@ export class CompilerImports extends Plugin {
-
-            if (exist) {
-              return provider.get(url, (error, content) => {
-                if (error) return reject(error)
-                resolve(content)
-              })
-            }
+  async resolveAndSave (url, targetPath) {
+    if (url.indexOf('remix_tests.sol') !== -1) return remixTests.assertLibCode
+    try {
+      const provider = await this.call('fileManager', 'getProviderOf', url)
+      if (provider) {
+        if (provider.type === 'localhost' && !provider.isConnected()) {
+          throw new Error(`file provider ${provider.type} not available while trying to resolve ${url}`)
+        }
+        const exist = await provider.exists(url)
+        /*
+          if the path is absolute and the file does not exist, we can stop here
+          Doesn't make sense to try to resolve "localhost/node_modules/localhost/node_modules/<path>" and we'll end in an infinite loop.
+        */
+        if (!exist && url.startsWith('browser/')) throw new Error(`not found ${url}`)
+        if (!exist && url.startsWith('localhost/')) throw new Error(`not found ${url}`)
 
-            // try to resolve localhost modules (aka truffle imports) - e.g from the node_modules folder
-            this.call('fileManager', 'getProviderByName', 'localhost').then((localhostProvider) => {
-              if (localhostProvider.isConnected()) {
-                var splitted = /([^/]+)\/(.*)$/g.exec(url)
-                return async.tryEach([
-                  (cb) => { this.resolveAndSave('localhost/installed_contracts/' + url, null).then((result) => cb(null, result)).catch((error) => cb(error.message)) },
-                  // eslint-disable-next-line standard/no-callback-literal
-                  (cb) => { if (!splitted) { cb('URL not parseable: ' + url) } else { this.resolveAndSave('localhost/installed_contracts/' + splitted[1] + '/contracts/' + splitted[2], null).then((result) => cb(null, result)).catch((error) => cb(error.message)) } },
-                  (cb) => { this.resolveAndSave('localhost/node_modules/' + url, null).then((result) => cb(null, result)).catch((error) => cb(error.message)) },
-                  // eslint-disable-next-line standard/no-callback-literal
-                  (cb) => { if (!splitted) { cb('URL not parseable: ' + url) } else { this.resolveAndSave('localhost/node_modules/' + splitted[1] + '/contracts/' + splitted[2], null).then((result) => cb(null, result)).catch((error) => cb(error.message)) } }],
-                (error, result) => {
-                  if (error) {
-                    return this.importExternal(url, targetPath, (error, content) => {
-                      if (error) return reject(error)
-                      resolve(content)
-                    })
-                  }
-                  resolve(result)
-                })
-              }
-              this.importExternal(url, targetPath, (error, content) => {
+        if (exist) {
+          const content = await (() => {
```

Comment:
This seems a bit complex code. May be it can be simplified

Reply:
The issue here is that the provider doesn't return a promise but uses callback, but we need this call to `await`...